### PR TITLE
Add July 2023 Randomized Format Spotlight

### PR DESCRIPTION
**Pull request by livid-washed**
This implements Random Roulette, which is the next Randomized Format Spotlight, replacing Mix and Mega Cup. https://www.smogon.com/forums/threads/randbats-spotlight-cycle-8-voting.3723432/

It chooses a random number between 1-9 inclusive and becomes a Random Battle in that gen.

### CODE REVIEW

File: data/mods/randomroulette/scripts.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+		// Generate teams using the format
+		for (const side of this.sides) {
+			this.teamGenerator.setSeed(PRNG.generateSeed());
+
+			const team = this.teamGenerator.getTeam();
+			side.team = team;
+			side.pokemon = [];
+			for (let i = 0; i < team.length && i < 24; i++) {
+				// console.log("NEW POKEMON: " + (this.team[i] ? this.team[i].name : '[unidentified]'));
```
Comment (AnnikaCodes) [inline data/mods/randomroulette/scripts.ts:27]:
```suggestion
```

**Review by AnnikaCodes — COMMENTED** [review]
Has this been tested? Should I deploy it on http://staging.psim.us/ first?

**Comment by livid-washed** [comment]
> Has this been tested? Should I deploy it on http://staging.psim.us/ first?

I've tested it locally and it works

File: config/formats.ts [hunk 2]
```diff
@@ -1917,35 +1917,10 @@ export const Formats: FormatList = [
...
-		onSwitchOut(pokemon) {
-			// @ts-ignore
-			const oMegaSpecies = this.dex.species.get(pokemon.species.originalSpecies);
-			if (oMegaSpecies.exists && pokemon.m.originalSpecies !== oMegaSpecies.baseSpecies) {
-				this.add('-end', pokemon, oMegaSpecies.requiredItem || oMegaSpecies.requiredMove, '[silent]');
-			}
-		},
+		name: "[Gen 9] Random Roulette",
+		desc: `Random Battles in a random generation! [Gen 1] Random Battle - [Gen 9] Random Battle.`,
+		mod: 'randomroulette',
```
Comment (KrisXV) [inline config/formats.ts:1922]:
```suggestion
		desc: `Random Battles in a random generation! [Gen 1] Random Battle - [Gen 9] Random Battle.`,

		mod: 'randomroulette',
```

**Pull request merged**